sdk/tests/conformance2/extensions/webgl-multi-draw-instanced-base-vertex-base-instance.html

```diff
@@ -269,9 +255,248 @@
+      if (!areArraysEqual(was, expected)) {
+        testFailed(`${subtest}: Expected [${expected}], was [${was}]. desc: ${JSON.stringify(desc)}`);
+      } else {
+        debug(`${subtest}: Was [${was}] as expected.`);
+      }
+    };
+  })();
+
+  // Common setup complete
+  // -
+  // Create testcases
+
+  const DRAW_FUNC_COMBINER = [{
+    name: 'drawArraysInstanced',
+    draw: desc => {
+      if (desc.base_vert) return false;
+      if (desc.base_inst) return false;
+      gl.drawArraysInstanced(gl[desc.mode], 0,
+          desc.vert_count, desc.inst_count);
+      return true;
+    },
+  }, {
+    name: 'drawElementsInstanced',
+    draw: desc => {
+      if (desc.base_vert) return false;
+      if (desc.base_inst) return false;
+      gl.drawElementsInstanced(gl[desc.mode], desc.vert_count,
+          gl.UNSIGNED_INT, 0, desc.inst_count);
+      return true;
+    },
+  }, {
+    name: 'drawArraysInstancedBaseInstanceWEBGL',
+    draw: desc => {
+      if (desc.base_vert) return false;
+      if (!EXT) return false;
+      EXT.drawArraysInstancedBaseInstanceWEBGL(gl[desc.mode],
+        desc.base_vert, desc.vert_count, desc.inst_count,
+        desc.base_inst);
+      return true;
+    },
+  }, {
+    name: 'drawElementsInstancedBaseVertexBaseInstanceWEBGL',
+    draw: desc => {
+      if (!EXT) return false;
+      EXT.drawElementsInstancedBaseVertexBaseInstanceWEBGL(
+          gl[desc.mode], desc.vert_count, gl.UNSIGNED_INT, 0,
+          desc.inst_count, desc.base_vert, desc.base_inst);
+      return true;
+    },
+  }];
+
+  // -
+
+  function make_key_combiner(key, vals) {
+    const ret = [];
+    for (const v of vals) {
+      const cur = {};
+      cur[key] = v;
+      ret.push(cur);
+    }
+    return ret;
+  }
+
+  const TEST_DESCS = crossCombine(
+    DRAW_FUNC_COMBINER,
+    make_key_combiner('base_vert', [0,1,2]),
+    make_key_combiner('vert_count', [0,1,2]),
+    make_key_combiner('base_inst', [0,1,2]),
+    make_key_combiner('inst_count', range(10)),
+  );
+  console.log('TEST_DESCS', TEST_DESCS);
+
+  // -
+  // Run testcases
+
+  gl.disable(gl.DEPTH_TEST);
+  gl.disable(gl.STENCIL_TEST);
+  gl.disable(gl.BLEND);
+
+  for (const desc of TEST_DESCS) {
+    gl.disable(gl.SCISSOR_TEST);
+    gl.clearBufferfv(gl.COLOR, 0, [1,0,0,1]);
+
+    const last_vert_id = desc.base_vert + desc.vert_count - 1;
+    const last_inst_id = desc.inst_count - 1;
+    const last_inst = desc.inst_count - 1;
+    const last_inst_div1 = desc.base_inst + last_inst;
+    const last_inst_div2 = desc.base_inst + Math.floor(last_inst / 2);
+    const last_inst_div3 = desc.base_inst + Math.floor(last_inst / 3);
+
+    gl.useProgram(vertid_prog);
+    if (!desc.draw(desc)) {
+      //debug('Not supported for draw func, skipping...');
+      continue;
+    }
+    debug('\ndesc: ' + JSON.stringify(desc));
+
+    wtu.glErrorAssert(gl, 0);
+    if (!desc.vert_count || !desc.inst_count) {
+      expect_pixel(desc, 'vertid_prog', [255, 0, 0, 255]);
```

Comment:
Q: when instanceCount is 0, is it a no-op? (testing on windows chrome this fails and I remember once read somewhere D3D treats instanceCount =0 and 1 the same? But likely I did it wrong)

Reply:
instanceCount=0 draws nothing, but I think it's not something we can early-out from (it can still generate errors), so I hesitate to call it a no-op.